mteb/tasks/Retrieval/tur/TurHistQuad.py

```diff
@@ -0,0 +1,96 @@
+import datasets
+
+from mteb.abstasks import AbsTaskRetrieval, TaskMetadata
+
+
+class TurHistQuadRetrieval(AbsTaskRetrieval):
+    metadata = TaskMetadata(
+        name="TurHistQuadRetrieval",
+        dataset={
+            "path": "asparius/TurHistQuAD",
+            "revision": "2a2b8ddecf1189f530676244d0751e1d0a569e03",
+        },
+        description="Question Answering dataset on Ottoman History in Turkish",
+        reference="https://github.com/okanvk/Turkish-Reading-Comprehension-Question-Answering-Dataset",
+        type="Retrieval",
+        category="p2p",
+        eval_splits=["test"],
+        eval_langs=["tur-Latn"],
+        main_score="ndcg_at_10",
+        date=("2021-01-01", "2021-10-13"),
+        form=["written"],
+        task_subtypes=["Question answering"],
+        domains=["Encyclopaedic", "Non-fiction", "Academic"],
+        license="MIT",
+        socioeconomic_status="high",
+        annotations_creators="derived",
+        dialect=[],
+        text_creation="found",
+        bibtex_citation="""
+            @INPROCEEDINGS{9559013,
+                author={Soygazi, Fatih and Çiftçi, Okan and Kök, Uğurcan and Cengiz, Soner},
+                booktitle={2021 6th International Conference on Computer Science and Engineering (UBMK)}, 
+                title={THQuAD: Turkish Historic Question Answering Dataset for Reading Comprehension}, 
+                year={2021},
+                volume={},
+                number={},
+                pages={215-220},
+                keywords={Computer science;Computational modeling;Neural networks;Knowledge discovery;Information retrieval;Natural language processing;History;question answering;information retrieval;natural language understanding;deep learning;contextualized word embeddings},
+                doi={10.1109/UBMK52708.2021.9559013}}
+
+        """,
+        n_samples={"test": 1330, "train": 14221},
+        avg_character_length={"train": 1219.37, "test": 1513.83},
+    )
+
+    def load_data(self, **kwargs) -> None:
+        """And transform to a retrieval datset, which have the following attributes
+
+        self.corpus = Dict[doc_id, Dict[str, str]] #id => dict with document datas like title and text
+        self.queries = Dict[query_id, str] #id => query
+        self.relevant_docs = Dict[query_id, Dict[[doc_id, score]]
+        """
+        if self.data_loaded:
+            return
+
+        self.dataset = datasets.load_dataset(**self.metadata_dict["dataset"])
+
+        self.corpus = {}
+        self.relevant_docs = {}
+        self.queries = {}
+        text2id = {}
+
+        for split in self.dataset:
```

Comment:
The evaluate method only consider `eval_splits`, so no need to go through this for the `train` split.